hil/api.py

```diff
@@ -116,10 +117,15 @@ def project_detach_node(project, node):
 
     If the node has network attachments or pending network actions, a
     BlockedError will be raised.
+
+    If using a maintenance pool, please update the hil config accordingly.
```

Comment:
This doesn't seem like a very good place for advice to sysadmins.

It might make sense to say something along the lines of "if a maintenance pool is configured, the maintenance service will be notified and the node will be moved to the maintenance project. otherwise it will be returned directly to the free pool."

...but we should probably put that in rest_api.md instead.